Extensions/PrimitiveDrawing/shapepainterruntimeobject.ts

```diff
@@ -438,23 +450,294 @@ namespace gdjs {
+        return;
+      }
+      this._scaleY = newScale * (this._flippedY ? -1 : 1);
+      this._renderer.updateScaleY();
+      this.hitBoxesDirty = true;
+    }
+
+    flipX(enable: boolean): void {
+      if (enable !== this._flippedX) {
+        this._scaleX *= -1;
+        this._flippedX = enable;
+        this._renderer.updateScaleX();
+        this.hitBoxesDirty = true;
+      }
+    }
+
+    flipY(enable: boolean): void {
+      if (enable !== this._flippedY) {
+        this._scaleY *= -1;
+        this._flippedY = enable;
+        this._renderer.updateScaleY();
+        this.hitBoxesDirty = true;
+      }
+    }
+
+    isFlippedX(): boolean {
+      return this._flippedX;
+    }
+
+    isFlippedY(): boolean {
+      return this._flippedY;
+    }
+
+    /**
+     * Get the scale of the object (or the geometric mean of the X and Y scale in case they are different).
+     *
+     * @return the scale of the object (or the geometric mean of the X and Y scale in case they are different).
+     */
+    getScale(): number {
+      return this._scaleX === this._scaleY
+        ? this._scaleX
+        : Math.sqrt(this._scaleX * this._scaleY);
+    }
+
+    /**
+     * Get the scale of the object on Y axis.
+     *
+     * @return the scale of the object on Y axis
+     */
+    getScaleY(): float {
+      return Math.abs(this._scaleY);
+    }
+
+    /**
+     * Get the scale of the object on X axis.
+     *
+     * @return the scale of the object on X axis
+     */
+    getScaleX(): float {
+      return Math.abs(this._scaleX);
+    }
+
+    invalidateBounds() {
+      this.hitBoxesDirty = true;
+    }
+
+    getDrawableX(): float {
+      return this._renderer.getDrawableX();
+    }
+
+    getDrawableY(): float {
+      return this._renderer.getDrawableY();
     }
 
     getWidth(): float {
-      return 32;
+      return this._renderer.getWidth();
     }
 
     getHeight(): float {
-      return 32;
+      return this._renderer.getHeight();
+    }
+
+    updatePreRender(runtimeScene: gdjs.RuntimeScene): void {
+      this._renderer.updatePreRender();
+    }
+
+    transformToDrawing(x: float, y: float) {
+      const point = ShapePainterRuntimeObject._pointForTransformation;
+      point[0] = x;
+      point[1] = y;
+      return this._renderer.transformToDrawing(point);
+    }
+
+    transformToScene(x: float, y: float) {
+      const point = ShapePainterRuntimeObject._pointForTransformation;
+      point[0] = x;
+      point[1] = y;
+      return this._renderer.transformToScene(point);
+    }
+
+    transformToDrawingX(x: float, y: float) {
+      return this.transformToDrawing(x, y)[0];
+    }
+
+    transformToDrawingY(x: float, y: float) {
+      return this.transformToDrawing(x, y)[1];
+    }
+
+    transformToSceneX(x: float, y: float) {
+      return this.transformToScene(x, y)[0];
+    }
+
+    transformToSceneY(x: float, y: float) {
+      return this.transformToScene(x, y)[1];
+    }
+
+    updateHitBoxes(): void {
```

Comment:
It would make it harder to understand the "absolute coordinates" impact. If we wanted to really rely on the base implementation we should probably use a protected method with an extra boolean but this is very specific to the ShapePainter so it will probably pollute the base class.
To over-engineering it, it could be abstract and implemented with an hitbox builder.

Reply:
Yep, I was thinking about a :

```
if (!absolute coordinates) { 
  super.updateHitBoxes(); // Use getDrawableX, getDrawableY, getWidth, getHeight 
} else { ... }
```

But might not be worth it.